core/src/main/java/com/exadel/etoolbox/linkinspector/core/services/util/LinkInspectorResourceUtil.java

```diff
@@ -59,6 +55,37 @@ public static void removeResource(String path, ResourceResolver resourceResolver
         }
     }
 
+    public static void createNode(String path, ResourceResolver resourceResolver) {
+        try {
+            Session session = resourceResolver.adaptTo(Session.class);
+            if (session == null) {
+                LOG.warn("Session is null, recreating node is interrupted");
+                return;
+            }
+            JcrUtil.createPath(path, JcrConstants.NT_UNSTRUCTURED, session);
+            session.save();
+        } catch (RepositoryException e) {
+            LOG.error(String.format("Failed to recreate node %s", path), e);
+        }
+    }
+
+    public static void addParamToNode(String path, ResourceResolver resourceResolver, String paramName, Long paramValue) {
+        try {
+            final Session session = resourceResolver.adaptTo(Session.class);
+            if (session == null) {
+                LOG.warn("Session is null, recreating node is interrupted");
+                return;
+            }
+            final Node node = session.getNode(path);
+            if (node != null) {
```

Comment:
(methodic) It is usually suggested to stick to one conceptual level of storage. 
Lowest is the JCR level. There live "Node"s. 
Next is Sling level. There live "Resource"s. 
Upper is WCM level. There live "Page"s, etc.

We usually operate at Sling level most of the time. So why go down to "Node" level? Just do like `resource.adaptTo(ModifiableValueMap.class).put(key, value)...`